frontend/src/state/AppState.jsx

```diff
@@ -114,11 +138,108 @@ export function useHierarchySettings() {
       }
       default:
         return state;
+      */
+  }
+  return {...state, ...graphOptions};
+}
+/*
+const GraphOptionsContext = createContext(null);
+export function GraphOptionsProvider({ children }) {    // settings 1
+  const initialSettings = {};
+  // const initialSettings = null;
+  /* {
+    // graphOptions will actually be initialized in GraphContainer.setGraphDisplayConfig
+    graphOptions: {
+      specialConceptTreatment: {},
+      nested: true,
+      // hideRxNormExtension: true,
+    }
+    // collapsePaths: {}, // collapsedDescendantPaths: {}, // hideZeroCounts: false,
+  }; * /
+  const storageProvider = useSearchParamsState();
+  let state = storageProvider.getItem('graphOptions') || initialSettings;
+
+  const dispatch = action => {
+    let latestState = storageProvider.getItem('graphOptions') || initialSettings;
+    const stateAfterDispatch = settingsReducer(latestState, action);
+    if (!isEqual(latestState, stateAfterDispatch)) {
+      storageProvider.setItem('graphOptions', stateAfterDispatch);
     }
   }
-  const [state, dispatch] = usePersistedReducer(hierarchySettingsReducer);
-  return [state, dispatch];
+  return (
+      <GraphOptionsContext.Provider value={[state, dispatch]}>
+        {children}
+      </GraphOptionsContext.Provider>
+  );
+}
+export function useGraphOptions() {
+  return useContext(GraphOptionsContext);
 }
+*/
+
+export const [GraphOptionsProvider, useGraphOptions] = makeProvider({
+  name: 'graphOptions', reducer: graphOptionsReducer, initialSettings: {}, storageProviderGetter: useSearchParamsState, });
+
+function makeProvider({name, reducer, initialSettings, storageProviderGetter, jsonify=false, forceUpdateForConsumers=false}) {
+  const nonUpdatingReducer = (state, action) => {
+    const newState = reducer(state, action);
+    return isEqual(state, newState) ? state : newState;
+  };
+
+  const Context = createContext();
+  const Provider = ({children}) => {
```

Comment:
Are `Context` and `Provider` new? 
Are you ready to discuss these or are they still in flux?

This is the first time I've seen `Context` or `Provider` by themselves, without some qualifier before it, like `MySuchAndSuchContext`.
I don't see how they're used yet either. Are they being used yet?